src/System.Management.Automation/engine/CommandCompletion/CompletionAnalysis.cs

```diff
@@ -505,6 +516,15 @@ internal List<CompletionResult> GetResultHelper(CompletionContext completionCont
                             replacementIndex = completionContext.ReplacementIndex;
                             replacementLength = completionContext.ReplacementLength;
                         }
+                        else if (lastAst.Parent is CommandExpressionAst
+                            && lastAst.Parent.Parent is AssignmentStatementAst assignmentAst2)
+                        {
+                            // Handle scenarios like '[ValidateSet(1,2)][int]$i = 2; $i = <tab>'
```

Comment:
This scenario should have a number token at the cursor, so I guess a better example would be:
```
'[ValidateSet(10,22)][int]$i = 10; $i = 2<tab>'
```

Reply:
goti t